src/core/components/documentParser/documentParser.tsx

```diff
@@ -0,0 +1,56 @@
+import { Image } from '@tiptap/extension-image';
+import TipTapLink from '@tiptap/extension-link';
+import { EditorContent, useEditor } from '@tiptap/react';
+import { StarterKit } from '@tiptap/starter-kit';
+import classNames from 'classnames';
+import { useEffect, type ComponentProps } from 'react';
+import sanitizeHtml from 'sanitize-html';
+import { Markdown } from 'tiptap-markdown';
+
+export interface IDocumentParserProps extends Omit<ComponentProps<'div'>, 'ref'> {
+    stringDocument: string;
```

Comment:
Let's always document the component properties. I'd also call it just `document`, we don't usually prefix props with their types